### PR TITLE
Update the dashboard client for ROS2

### PR DESCRIPTION
**Pull request by AndyZe**

### CODE REVIEW

File: ur_robot_driver/src/dashboard_client_ros.cpp [hunk 1]
```diff
@@ -74,6 +74,17 @@ DashboardClientROS::DashboardClientROS(const rclcpp::Node::SharedPtr& node, cons
 
   // Stop program execution on the robot
   stop_service_ = createDashboardTriggerSrv("stop", "stop\n", "Stopped");
+
+  // Dismiss a protective stop to continue robot movements. NOTE: It is the responsibility of the user to ensure the
+  // cause of the protective stop is resolved before calling this service.
+  unlock_protective_stop_service_ =
+      createDashboardTriggerSrv("unlock_protective_stop", "unlock protective stop\n", "Protective stop releasing");
+
+  using namespace std::placeholders;
```
Comment (AndyZe) [inline ur_robot_driver/src/dashboard_client_ros.cpp:83]: I'd prefer not to have this here. It's just not something I've seen very often. Go ahead and spell it out completely:

std::placeholders::_1

Comment (AndyZe) [reply ur_robot_driver/src/dashboard_client_ros.cpp:83]: @tylerjw do you agree with me?  (Tyler is our C++ guru)

Comment (tylerjw) [reply ur_robot_driver/src/dashboard_client_ros.cpp:83]: This is sort of a special case.  The placeholders are annoying and this is nowhere near as bad as `using namespace std;`.

Comment (livanov93) [reply ur_robot_driver/src/dashboard_client_ros.cpp]: @AndyZe  I will remove it for now, and leave the full spelling e.g. `std::placeholders::_1, std::placeholders::_2 `

File: ur_robot_driver/include/ur_robot_driver/dashboard_client_ros.h [hunk 2]
```diff
@@ -51,5 +56,80 @@ namespace ur_robot_driver
...
+  rclcpp::Service<ur_dashboard_msgs::srv::IsProgramRunning>::SharedPtr running_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::GetLoadedProgram>::SharedPtr get_loaded_program_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::IsProgramSaved>::SharedPtr is_program_saved_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::Popup>::SharedPtr popup_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::GetProgramState>::SharedPtr program_state_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::GetSafetyMode>::SharedPtr safety_mode_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::GetRobotMode>::SharedPtr robot_mode_service_;
+
+  // EXISTS AS SPARE IN ROS1 DRIVER
+  //    rclcpp::Service<ur_dashboard_msgs::srv::> polyscope_version_service_;
```
Comment (AndyZe) [inline ur_robot_driver/include/ur_robot_driver/dashboard_client_ros.h:133]: I don't understand what this is. Why not just delete it?

File: ur_robot_driver/src/dashboard_client_ros.cpp [hunk 3]
```diff
@@ -29,4 +29,324 @@
...
+      "get_robot_mode",
+      std::bind(&DashboardClientROS::handleRobotModeQuery, this, std::placeholders::_1, std::placeholders::_2));
+
+  // Service to add a message to the robot's log
+  add_to_log_service_ = node->create_service<ur_dashboard_msgs::srv::AddToLog>(
+      "add_to_log", [&](const ur_dashboard_msgs::srv::AddToLog::Request::SharedPtr req,
+                        ur_dashboard_msgs::srv::AddToLog::Response::SharedPtr resp) {
+        resp->answer = this->client_.sendAndReceive("addToLog " + req->message + "\n");
+        resp->success = std::regex_match(resp->answer, std::regex("(Added log message|No log message to add)"));
+
```
Comment (AndyZe) [inline ur_robot_driver/src/dashboard_client_ros.cpp:168]: let's delete this empty line, just to match the others

**Review by AndyZe — COMMENTED** [review]
Looks good to me! Just a couple of tiny comments to fix before merging

File: ur_robot_driver/include/ur_robot_driver/dashboard_client_ros.h (hunk 2 above)
Comment (livanov93) [reply ur_robot_driver/include/ur_robot_driver/dashboard_client_ros.h:133]: It existed in the old driver. It is left for possible UR's idea implementation.

**Pull request merged**